cmd/config-current-notify.go

```diff
@@ -0,0 +1,936 @@
+	notifyESSubSys       = "notify.elasticsearch"
+	notifyAMQPSubSys     = "notify.amqp"
+	notifyPostgresSubSys = "notify.postgres"
+	notifyRedisSubSys    = "notify.redis"
+	notifyWebhookSubSys  = "notify.webhook"
+
+	// Add new constants here if you add new fields to config.
+)
+
+func (s serverConfig) SetNotifyKafka(kName string, cfg target.KafkaArgs) error {
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	s[notifyKafkaSubSys][kName] = []KV{
+		{
+			Key: stateKey,
+			Value: func() string {
+				if cfg.Enable {
+					return stateEnabled
+				}
+				return stateDisabled
+			}(),
+		},
+		{
+			Key: "brokers",
+			Value: func() string {
+				var brokers []string
+				for _, broker := range cfg.Brokers {
+					brokers = append(brokers, broker.String())
+				}
+				return strings.Join(brokers, ",")
+			}(),
+		},
+		{
+			Key:   "topic",
+			Value: cfg.Topic,
+		},
+		{
+			Key:   "queueDir",
+			Value: cfg.QueueDir,
+		},
+		{
+			Key:   "queueLimit",
+			Value: strconv.Itoa(int(cfg.QueueLimit)),
+		},
+		{
+			Key:   "tls.enable",
+			Value: strconv.FormatBool(cfg.TLS.Enable),
+		},
+		{
+			Key:   "tls.skipVerify",
+			Value: strconv.FormatBool(cfg.TLS.SkipVerify),
+		},
+		{
+			Key:   "tls.clientAuth",
+			Value: strconv.Itoa(int(cfg.TLS.ClientAuth)),
+		},
+		{
+			Key:   "sasl.enable",
+			Value: strconv.FormatBool(cfg.SASL.Enable),
+		},
+		{
+			Key:   "sasl.user",
+			Value: cfg.SASL.User,
+		},
+		{
+			Key:   "sasl.password",
+			Value: cfg.SASL.Password,
+		},
+	}
+	return nil
+}
+
+func (s serverConfig) GetNotifyKafka() map[string]target.KafkaArgs {
+	kafkaTargets := make(map[string]target.KafkaArgs)
+	for k, kvs := range s[notifyKafkaSubSys] {
+		enabled := kvs.Get(stateKey) == stateEnabled
+		if !enabled {
+			continue
+		}
+		var err error
+		var brokers []xnet.Host
+		for _, s := range strings.Split(kvs.Get("brokers"), ",") {
+			var host *xnet.Host
+			host, err = xnet.ParseHost(s)
+			if err != nil {
+				break
+			}
+			brokers = append(brokers, *host)
+		}
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+
+		queueLimit, err := strconv.Atoi(kvs.Get("queueLimit"))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+
+		clientAuth, err := strconv.Atoi(kvs.Get("tls.clientAuth"))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+
+		kafkaArgs := target.KafkaArgs{
+			Enable:     enabled,
+			Brokers:    brokers,
+			Topic:      kvs.Get("topic"),
+			QueueDir:   kvs.Get("queueDir"),
+			QueueLimit: uint16(queueLimit),
+		}
+
+		kafkaArgs.TLS.Enable = kvs.Get("tls.enable") == "true"
+		kafkaArgs.TLS.SkipVerify = kvs.Get("tls.skipVerify") == "true"
+		kafkaArgs.TLS.ClientAuth = tls.ClientAuthType(clientAuth)
+
+		kafkaArgs.SASL.Enable = kvs.Get("sasl.enable") == "true"
+		kafkaArgs.SASL.User = kvs.Get("sasl.user")
+		kafkaArgs.SASL.Password = kvs.Get("sasl.password")
+
+		if err = kafkaArgs.Validate(); err != nil {
```

Comment:
Why do we need to validate the targets in get? I see we're already validating in set. Or is this some sort of integrity check?

Reply:
We need to validate what we read from the disk @Praveenrajmani - since these are just strings.

Reply:
ok